packages/apps/desktop-viewer-test/src/frontend/components/frontstages/IModelSelectFrontstage.tsx

```diff
@@ -0,0 +1,91 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+import { IModelInfo, IModelSelector } from "@bentley/imodel-select-react";
+import {
+  BackstageAppButton,
+  ConfigurableCreateInfo,
+  ContentControl,
+  ContentGroup,
+  ContentLayoutDef,
+  CoreTools,
+  Frontstage,
+  FrontstageProps,
+  FrontstageProvider,
+  ToolWidgetComposer,
+  Widget,
+  Zone,
+} from "@bentley/ui-framework";
+import * as React from "react";
+
+import { ITwinViewerApp } from "../../app/ITwinViewerApp";
+
+/* eslint-disable react/jsx-key */
+
+class IModelSelectorControl extends ContentControl {
+  constructor(info: ConfigurableCreateInfo, options: any) {
+    super(info, options);
+
+    this.reactNode = <IModelSelector onIModelSelected={this._onSelectIModel} />;
+  }
+
+  // called when an imodel has been selected on the IModelSelect
+  private _onSelectIModel = async (iModelInfo: IModelInfo) => {
+    ITwinViewerApp.store.dispatch({
+      type: "App:OPEN_IMODEL",
+      payload: {
+        projectName: iModelInfo.projectInfo.name,
```

Comment:
+1 to my other comment about if the names are even necessary

Reply:
removed